### PR TITLE
KAFKA-12809: Remove Deprecated methods under Stores

### PR DESCRIPTION
**Pull request by jlprat**
Removes deprecated methods since 2.1
Moves needed implementation from deprecated method to the right new one.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/state/Stores.java [hunk 1]
```diff
@@ -364,38 +324,22 @@ public static WindowBytesStoreSupplier inMemoryWindowStore(final String name,
...
      * @return an instance of a {@link  SessionBytesStoreSupplier}
-     * @deprecated since 2.1 Use {@link Stores#persistentSessionStore(String, Duration)} instead
      */
-    @Deprecated // continuing to support Windows#maintainMs/segmentInterval in fallback mode
     public static SessionBytesStoreSupplier persistentSessionStore(final String name,
-                                                                   final long retentionPeriodMs) {
+                                                                   final Duration retentionPeriod) {
+        final String msgPrefix = prepareMillisCheckFailMsgPrefix(retentionPeriod, "retentionPeriod");
+        final long retentionPeriodMs = validateMillisecondDuration(retentionPeriod, msgPrefix);
         Objects.requireNonNull(name, "name cannot be null");
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/state/Stores.java]: We should do this check first

Comment (jlprat) [reply streams/src/main/java/org/apache/kafka/streams/state/Stores.java]: Fixed

**Review by mjsax — APPROVED** [review]
Thanks. LGTM. Will merge after Jenkins passed.

**Comment by mjsax** [comment]
Merged to `trunk`. Thanks for the PR @jlprat!

**Comment by jlprat** [comment]
Thanks for the review

**Pull request merged**